src/components/VideoPreview.tsx

```diff
@@ -68,6 +70,11 @@ export const VideoPreviewPublisher: React.FC<VideoPreviewWPublisherProps> = ({ i
     video?.channel.avatarPhotoDataObject
   )
 
+  const hasThumbnailUploadFailed =
+    video?.thumbnailPhotoAvailability === AssetAvailability.Pending &&
+    uploadsState.flat().find((asset) => asset.ipfsContentId === video.thumbnailPhotoDataObject?.ipfsContentId)
+      ?.lastStatus !== 'completed'
```

Comment:
I think we shouldn't need to rely on `uploadsState`, I don't like this coupling much. Wouldn't just checking if the asset is pending work?

Reply:
At the beginning it relied only on `AssetAvailability`, but then discovered some cases where this didn't quite work. Will go back to that and double-check that

Reply:
should be working now without `uploadsState`